### PR TITLE
Add Sparkline component and replace tiny-linked-charts dependency

### PR DESCRIPTION
**Pull request by hughess**
### Description
Adds a Sparkline component and replaces the existing `svelte-tiny-linked-charts` dependency, which caused some issues with rendering, dependency management, and stability in the past.

As a result, this replaces the sparkline included with BigValue. This can also fit into DataTable in the future.

This uses echarts under the hood with some customization. It includes:
- SSR
- Hydration to interactive if `interactive=true`
- Tooltip above and below the sparkline
- 3 chart types: line, area, and bar
- `connectGroup` prop which will synchronize tooltip behaviour across any sparklines where the `connectGroup` name matches

#### Basic sparkline
sparkline-basic

```html
<Sparkline 
    data={sales_by_date} 
    dateCol=date 
    valueCol=sales 
/>
```

#### Connected sparklines
sparkline-connected

```html
<Sparkline data={sales_by_date} dateCol=date valueCol=sales type=bar  valueFmt=eur dateFmt=mmm connectGroup=mysparkline/>
<Sparkline data={sales_by_date} dateCol=date valueCol=sales type=area color=maroon valueFmt=eur dateFmt=mmm connectGroup=mysparkline/>
<Sparkline data={sales_by_date} dateCol=date valueCol=sales type=line color=purple valueFmt=eur dateFmt=mmm connectGroup=mysparkline/>
```

#### In BigValue
CleanShot 2024-02-25 at 15 48 55

```html
<BigValue 
   data = {owc} 
   value=sales_usd0k 
   sparkline=month
   comparison=sales_change_pct0
   comparisonTitle="vs. Last Month"
   sparklineType=area
   connectGroup=mysparkline
/>

<BigValue data = {owc} 
   value=num_orders_num0
   title="Orders"
   sparkline=month
   sparklineType=bar
   comparison=num_orders_change_pct0
   comparisonTitle="vs. Last Month"
   sparklineColor=maroon
   sparklineDateFmt=shortdate
   connectGroup=mysparkline
/>

<BigValue data = {owc} 
   value=aov_usd2
   title="AOV ($)"
   sparkline=month
   comparison=aov_change_pct0
   comparisonTitle="vs. Last Month"
   sparklineColor=navy
   sparklineDateFmt=mmm
   sparklineYScale=true
   connectGroup=mysparkline
/> 
```

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable

### CODE REVIEW

File: packages/core-components/src/lib/unsorted/viz/core/_Sparkline.svelte
Comment (mcrascal) [inline packages/core-components/src/lib/unsorted/viz/core/_Sparkline.svelte:388]: The sparklines are visible through the mobile sidebar, suspect the z-50 is to blame

Comment (hughess) [reply packages/core-components/src/lib/unsorted/viz/core/_Sparkline.svelte:388]: Nice, good catch

File: packages/core-components/src/lib/unsorted/viz/core/_Sparkline.svelte
Comment (ItsMeBrianD) [inline packages/core-components/src/lib/unsorted/viz/core/_Sparkline.svelte:31]: Open issues for `DEFERRING` items, remove this comment

File: packages/core-components/src/lib/unsorted/viz/core/_Sparkline.svelte
Comment (ItsMeBrianD) [inline packages/core-components/src/lib/unsorted/viz/core/_Sparkline.svelte:85]: Can this logic live in a `.js` file rather than a large reactive statement?

**Review by ItsMeBrianD — CHANGES_REQUESTED** [review]
Looks mostly good, want to see if we can move the large reactive statement out of the .svelte file, that's caused a lot of issues trying to understand and maintain `Chart.svelte`

**Comment by archiewood** [comment]
What is the status of this PR?

**Comment by hughess** [comment]
should be good to go - just needs an approval!

**Review by archiewood — APPROVED** [review]

**Comment by archiewood** [comment]
And now it has one!

**Comment by hughess** [comment]
Looks like @ItsMeBrianD needs to approve after requesting changes

**Comment by mcrascal** [comment]
@hughess These are great! Two small nits below. I don't think we necessarily need to address these in this PR though.    

1. For inline sparklines, the interactive tooltip/label crowds the text around it (see below). I feel like this would matter less if interactive wasn't the default. 
2. I get a "strobing" effect on the red bar sparkline in the bigvalue example on mobile - I think it's because the gaps between the bars are extremely thin. Is there a way to remove the gap between the bars once you have a certain number of points? 


CleanShot 2024-02-28 at 17 30 56

**Comment by hughess** [comment]
Adjusted the tooltip to avoid the text overlap. 

CleanShot 2024-02-28 at 20 57 20

The bars will take more fiddling, so going to leave those for now - I also think part of that visual effect is coming from the saturation of the colour I put in there which is just a CSS colour (maroon).

I also just noticed another issue which I'm leaving for now - on page refresh, inline sparklines cause the text to shift before resetting. Going to see if this happens in a deployed site as well.

**Pull request merged**
